### DISCUSSION TITLE
Filters and Sorting in header

### DISCUSSION THREAD

**Discussion by raihanoza** [Comcast/react-data-grid#3676] [General]
I have filters and sorting in the header, but when I click the filter or sorting, the header is also selected (focus/selected), I want it not to be selected, how can I do that?

**Comment by paustint** [comment]
I think I ended up doing something like the below. It was a couple years ago when I implemented this, so cannot remember all the specifics.

```html
<div
      onClick={(ev) => ev.stopPropagation()}
      onPointerDown={(ev) => ev.stopPropagation()}
      onKeyDown={(ev) => ev.stopPropagation()}
>
...
</div>
```

<img width="718" alt="image" src="https://github.com/user-attachments/assets/56b59ffe-6478-498e-85bd-3b614f6d34f4" />

**Reply by raihanoza** [reply]
thanks for the response, but my header looks like this, the filter is directly in the header

![image](https://github.com/user-attachments/assets/eb55d0bb-ddfe-4d82-99d4-e00d6b31fb2a)

**Reply by paustint** [reply]
ok, regardless of what it looks like - did you try to make sure that stopped propagation for the following events to ensure that they do not bubble up to the grid event handlers?
For your filters, you want to make sure that you own these events and prevent them for making it to react grid.
```
      onClick={(ev) => ev.stopPropagation()}
      onPointerDown={(ev) => ev.stopPropagation()}
      onKeyDown={(ev) => ev.stopPropagation()}
```

**Comment by raihanoza** [comment]
some one help me please
